test/apps/iodemo/io_demo.cc

```diff
@@ -874,34 +916,27 @@ class DemoClient : public P2pDemoCommon {
     }
 
     void wait_for_responses(long max_outstanding) {
-        struct timeval tv_start = {};
-        bool timer_started      = false;
-        bool timer_finished     = false;
-        struct timeval tv_curr, tv_diff;
-        long count;
+        bool timer_started  = false;
+        bool timer_finished = false;
+        double start_time, curr_time, elapsed_time;
 
-        count = 0;
         while (((_num_sent - _num_completed) > max_outstanding) &&
                (_status == OK)) {
-            if ((count < 1000) || timer_finished) {
-                progress();
-                ++count;
+            progress();
+            if (timer_finished) {
                 continue;
             }
 
-            count = 0;
-
-            gettimeofday(&tv_curr, NULL);
+            curr_time = get_time();
 
             if (!timer_started) {
-                tv_start      = tv_curr;
+                start_time    = curr_time;
                 timer_started = true;
                 continue;
             }
 
-            timersub(&tv_curr, &tv_start, &tv_diff);
-            double elapsed = tv_diff.tv_sec + (tv_diff.tv_usec * 1e-6);
-            if (elapsed > _test_opts.client_timeout) {
+            elapsed_time = curr_time - start_time;
+            if (elapsed_time > _test_opts.client_timeout) {
```

Comment:
why need to change client side in this PR?

Reply:
there is a set of changes: 
1) short loop (< 1000) of progress was moved to common progress
2) `get_time` was moved to common class to be reused by server
3) since near code was changed, I noticed the code duplication of related code here and fixed it